osf/models/node.py

```diff
@@ -1811,13 +1811,21 @@ def register_node(self, schema, auth, data, parent=None):
 
         for node_relation in original.node_relations.filter(child__is_deleted=False):
             node_contained = node_relation.child
+            if child_ids and node_contained._id not in child_ids:
+                children = node_contained.node_relations.filter(child__is_deleted=False, child__guids___id__in=child_ids)
+                if children:  # We can't skip a node with children that we have to register.
```

Comment:
This should be `children.exists()`.

This will work, but it will execute slower SQL.

```python

In [1]: qs = OSFUser.objects.filter(username='doesnotexist')

In [2]: qs.exists()
SELECT (1) AS "a"
FROM "osf_osfuser"
WHERE "osf_osfuser"."username" = 'doesnotexist'
LIMIT 1

Execution time: 0.007378s [Database: default]

Out[2]: False

In [3]: bool(qs)
SELECT "osf_osfuser"."id",
       "osf_osfuser"."password",
       "osf_osfuser"."last_login",
       "osf_osfuser"."is_superuser",
       "osf_osfuser"."created",
       "osf_osfuser"."modified",
       "osf_osfuser"."content_type_pk",
       "osf_osfuser"."username",
       "osf_osfuser"."fullname",
       "osf_osfuser"."is_registered",
       "osf_osfuser"."is_claimed",
       "osf_osfuser"."security_messages",
       "osf_osfuser"."is_invited",
       "osf_osfuser"."unclaimed_records",
       "osf_osfuser"."contributor_added_email_records",
       "osf_osfuser"."merged_by_id",
       "osf_osfuser"."verification_key",
       "osf_osfuser"."verification_key_v2",
       "osf_osfuser"."email_last_sent",
       "osf_osfuser"."change_password_last_attempt",
       "osf_osfuser"."old_password_invalid_attempts",
       "osf_osfuser"."email_verifications",
       "osf_osfuser"."mailchimp_mailing_lists",
       "osf_osfuser"."osf_mailing_lists",
       "osf_osfuser"."date_registered",
       "osf_osfuser"."given_name",
       "osf_osfuser"."middle_names",
       "osf_osfuser"."family_name",
       "osf_osfuser"."suffix",
       "osf_osfuser"."external_identity",
       "osf_osfuser"."jobs",
       "osf_osfuser"."schools",
       "osf_osfuser"."social",
       "osf_osfuser"."date_last_login",
       "osf_osfuser"."date_confirmed",
       "osf_osfuser"."date_disabled",
       "osf_osfuser"."deleted",
       "osf_osfuser"."comments_viewed_timestamp",
       "osf_osfuser"."timezone",
       "osf_osfuser"."locale",
       "osf_osfuser"."requested_deactivation",
       "osf_osfuser"."notifications_configured",
       "osf_osfuser"."accepted_terms_of_service",
       "osf_osfuser"."is_active",
       "osf_osfuser"."is_staff",

  (SELECT JSON_AGG(JSON_BUILD_ARRAY(U0."modified", U0."id", U0."_id", U0."content_type_id", U0."object_id", U0."created")
                   ORDER BY U0."created" DESC) AS "__fields"
   FROM "osf_guid" U0
   WHERE (U0."object_id" = osf_osfuser."id")
     AND (U0."content_type_id" = 18)) AS "__guids"
FROM "osf_osfuser"
WHERE "osf_osfuser"."username" = 'doesnotexist'

Execution time: 0.011188s [Database: default]

Out[3]: False

```